### PR TITLE
Feb 2023 updates

### PR DESCRIPTION
**Pull request by RecursiveHook**
- Updates requirements.txt dependencies across the board
- Enables Discord API v9 support (Threads)
- Also updated the Heroku dynos to Ubu 22

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
 # API's
-discord.py[voice]==2.0.1
+discord.py[voice]==2.1.1
 
 # CHARTS
 matplotlib>=3.1.1
 pandas>=1.0.0
-emoji==0.5.4
+emoji==2.2.0
```
Comment (OxyOCE) [inline requirements.txt:20]: That's a big update - hopefully no API changes?

File: requirements.txt [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 
 # TEXT FORMATTING
-titlecase==0.12.0
+titlecase==2.4.0
```
Comment (OxyOCE) [inline requirements.txt:23]: Same here

File: requirements.txt [hunk 3]
```diff
@@ -29,15 +29,14 @@ jishaku==2.5.1
...
 # aiopg==1.0.0
 
 # CACHE
 # https://github.com/argaen/aiocache
-ujson==5.2.0
-msgpack==0.6.2
-# aiocache[redis,memcached]==0.10.1
-aiocache[redis,memcached]==0.11.1
+ujson==5.7.0
+msgpack==1.0.4
```
Comment (OxyOCE) [inline requirements.txt:38]: Same here

**Review by OxyOCE — APPROVED** [review]
LGTM, just a couple of big updates for requirements to keep an eye on in case there are any breaking changes to them.

:shipit:

File: requirements.txt (hunk 1 above)
Comment (RecursiveHook) [reply requirements.txt:20]: So about that...

```python
2023-02-12T00:34:48.183915+00:00 app[worker.1]: Ignoring exception in command poll-results:
2023-02-12T00:34:48.184614+00:00 app[worker.1]: Traceback (most recent call last):
2023-02-12T00:34:48.184765+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.11/site-packages/discord/ext/commands/core.py", line 229, in wrapped
2023-02-12T00:34:48.184765+00:00 app[worker.1]: ret = await coro(*args, **kwargs)
2023-02-12T00:34:48.184765+00:00 app[worker.1]: ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-02-12T00:34:48.184767+00:00 app[worker.1]: File "/app/cogs/polls.py", line 207, in collate_poll
2023-02-12T00:34:48.184767+00:00 app[worker.1]: for _, desc, reaction in gen_poll_options(poll):
2023-02-12T00:34:48.184783+00:00 app[worker.1]: File "/app/cogs/polls.py", line 54, in gen_poll_options
2023-02-12T00:34:48.184783+00:00 app[worker.1]: key = emoji.emojize(option[:option.find(' ')], use_aliases=True)
2023-02-12T00:34:48.184783+00:00 app[worker.1]: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-02-12T00:34:48.184802+00:00 app[worker.1]: TypeError: emojize() got an unexpected keyword argument 'use_aliases'
```

File: requirements.txt (hunk 2 above)
Comment (RecursiveHook) [reply requirements.txt:23]: Tested, works fine.

For ref, `titlecase` is used in Autorole for Other Games roles display.

File: requirements.txt (hunk 1 above)
Comment (RecursiveHook) [reply requirements.txt:20]: Tight, just needed to swap `use_aliases=True` for `language='alias'`.

**Pull request merged**